### PR TITLE
Unmaksed errors where applicable

### PR DESCRIPTION
**Pull request by kunagpal**
* Added preexisting help text as a help key within the original `error` object

### CODE REVIEW

File: examples/find-unique-urls-in-run.js [hunk 1]
```diff
@@ -13,7 +13,10 @@ newman.run({ collection: require('./sample-collection.json') })
         console.log(`Running ${args.cursor.length} request(s) and ${args.cursor.cycles} iteration(s)`);
     })
     .on('request', function (err, args) {
-        if (err) { return; }
+        if (err) {
```
Comment (shamasis) [inline examples/find-unique-urls-in-run.js]: not needed

File: lib/cli/index.js [hunk 2]
```diff
@@ -322,8 +322,7 @@ var _ = require('lodash'),
             result = parser.parseArgs(rawArgs.argv);
         }
         catch (e) {
-            e.help = parser.formatHelp();
-            return callback(e);
+            return callback(_.merge(e, { help: parser.formatHelp() }));
```
Comment (shamasis) [inline lib/cli/index.js]: where is this `formatHelp` function defined?

File: lib/reporters/cli/cli-utils.js [hunk 3]
```diff
@@ -95,7 +95,10 @@ cliUtils = {
             height;
 
         try { tty = require('tty'); }
-        catch (e) { tty = null; }
+        catch (e) {
+            console.error(e);
```
Comment (shamasis) [inline lib/reporters/cli/cli-utils.js]: not needed. this is for graceful fallback

File: lib/reporters/progress.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ var progress = require('cli-progress'),
     ProgressReporter;
 
 /**
- * Littlt reporter that generates a collection progress status bar on CLI
+ * Little reporter that generates a collection progress status bar on CLI
```
Comment (shamasis) [inline lib/reporters/progress.js]: 👍

File: lib/util.js [hunk 5]
```diff
@@ -37,7 +37,7 @@ module.exports = {
                     _.isString(body) && (body = parseJson(body));
                 }
                 catch (e) {
-                    return callback(new Error('Given URL did not provide a valid JSON response: ', location));
+                    return callback(_.merge(e, { help: `The URL: ${location} did not provide valid JSON data.` }));
```
Comment (shamasis) [inline lib/util.js]: use _.set

File: lib/util.js [hunk 6]
```diff
@@ -49,7 +49,7 @@ module.exports = {
                     value = parseJson(value.toString());
                 }
                 catch (e) {
-                    return callback(new Error('Given file does not contain valid JSON data: ', location));
+                    return callback(_.merge(e, { help: `The file at ${location} does not contain valid JSON data.` }));
```
Comment (shamasis) [inline lib/util.js]: use _.set

File: lib/cli/index.js (hunk 2 above)
Comment (kunagpal) [inline lib/cli/index.js]: This comes from `argparse`.

Comment (shamasis) [inline lib/cli/index.js]: @kunagpal

Comment (shamasis) [inline lib/cli/index.js]: @czardoz is this the way this will be done?

File: lib/config/rc-file.js [hunk 7]
```diff
@@ -24,14 +24,14 @@ module.exports.load = (callback) => {
...
-                return cb(null, {});
+                return cb(null, {}); // err masked to avoid overpopulating terminal with missing .newmanrc messages
             }
             data && data.toString && (data = data.toString());
             try {
                 return cb(null, parseJson(data));
             }
             catch (e) {
-                return cb(e); // File exists but contains invalid data. That's a fatal error.
+                return cb(_.merge(e, { help: `The file at ${path} contains invalid data.` }));
```
Comment (shamasis) [inline lib/config/rc-file.js]: _.set

**Pull request merged**